### PR TITLE
[BLACK-1092] E2E tests for Piecewise Linear NSD and FF

### PR DESCRIPTION
**Pull request by luigibertaco**
Ticket: https://q-ctrl.atlassian.net/browse/BLACK-1092

Blocked by: https://github.com/qctrl/jobs/pull/137 and https://github.com/qctrl/python/pull/72

Included tests for Piecewise Linear Noise Spectral Density validations.

Included tests for Filter Functions with a Piecewise Linear Noise Spectral Density.

### CODE REVIEW

File: tests/test_filter_functions.py [hunk 1]
```diff
@@ -115,13 +112,122 @@ def test_partial_calculate_filter_function(qctrl, system):
...
+            "duration": 0.16666666666666669,
+            "value": np.complex128(6.283185307179586 + 0j),
+        },
+    ]
+
+    DriveFactory.create(
+        name="Microwave",
+        operator=SIGMA_M / 2,
+        system=system,
+        pulse__type="custom",
```
Comment (saltysealion) [inline tests/test_filter_functions.py]: @luigibertaco we advocate using factories, lets use them in the test too so users who read them will follow.

File: tests/test_filter_functions.py [hunk 2]
```diff
@@ -115,13 +112,122 @@ def test_partial_calculate_filter_function(qctrl, system):
...
     assert filter_function.error_precision is None
 
+
+def test_calculate_filter_function_linear_nsd(qctrl, system):
+    DriveFactory = qctrl.factories.get_class("DriveFactory")
+    ShiftFactory = qctrl.factories.get_class("ShiftFactory")
+    qctrl.factories.get_class("DriftFactory")
+    AdditiveNoiseFactory = qctrl.factories.get_class("AdditiveNoiseFactory")
+    FilterFunctionFactory = qctrl.factories.get_class("FilterFunctionFactory")
+    NoiseSpectralDensityFactory = qctrl.factories.get_class(
```
Comment (saltysealion) [inline tests/test_filter_functions.py]: The names are long but is it worth using `PiecewiseLinearNoiseSpectralDensityFactory` specifically here? It is explicit and removes ambiguity around which factory is actually in use given there is also a factory named `NoiseSpectralDensityFactory`

File: tests/test_filter_functions.py (hunk 1 above)
Comment (luigibertaco) [reply tests/test_filter_functions.py]: 👍

File: tests/test_filter_functions.py (hunk 2 above)
Comment (luigibertaco) [reply tests/test_filter_functions.py]: 👍

**Comment by saltysealion** [comment]
✅

**Pull request merged**